exercises/react/src/example.c

```diff
@@ -41,42 +43,54 @@ struct reactor *create_reactor()
    return calloc(1, sizeof(struct reactor));
 }
 
-void destroy_reactor(struct reactor *r)
+static void destroy_cell(struct cell *c)
 {
-   struct child *child = r->first;
-   while (child) {
-      struct cb *cb = child->cell->cb;
-      while (cb) {
-         struct cb *next_cb = cb->next;
-         free(cb);
-         cb = next_cb;
-      }
-      free(child->cell);
+   struct cb *cb = c->cb;
+   while (cb) {
+      struct cb *next_cb = cb->next;
+      free(cb);
+      cb = next_cb;
+   }
 
+   struct child *child = c->child;
+   while (child) {
       struct child *next = child->next;
+      if (c == child->cell->input1) {
+         // Don't double-free for a compute2 cell.
```

Comment:
That was my main concern with making this. I thought it would be very tricky to free every cell exactly once, but every compute cell has exactly one `input1`, so it works out.